docs/Using-Fleet/configuration-files/README.md

```diff
@@ -264,15 +270,928 @@ spec:
+
+The `features` section of the configuration YAML lets you define what predefined queries are sent to the hosts and later on processed by Fleet for different functionalities.
+
+> Note: this section used to be named `host_settings`, but was renamed in Fleet v4.20.0,
+> `host_settings` is still supported for backwards compatibility.
+
+#### features.additional_queries
+
+Additional information to collect from hosts along with the host details. This information will be updated at the same time as other host details and is returned by the API when host objects are returned. Users must take care to keep the data returned by these queries small in order to mitigate potential performance impacts on the Fleet server.
+
+- Optional setting (dictionary of key-value strings).
+- Default value: none (empty).
+- Config file format:
+  ```
+  features:
+  	additional_queries:
+      time: SELECT * FROM time
+      macs: SELECT mac FROM interface_details
+  ```
+- Deprecated config file format:
+  ```
+  host_settings:
+  	additional_queries:
+      time: SELECT * FROM time
+      macs: SELECT mac FROM interface_details
+  ```
+
+#### features.enable_host_users
+
+Whether or not Fleet sends the query needed to gather user-related data from hosts.
+
+- Optional setting (boolean).
+- Default value: `true`.
+- Config file format:
+  ```
+  features:
+  	enable_host_users: false
+  ```
+- Deprecated config file format:
+  ```
+  host_settings:
+  	enable_host_users: false
+  ```
+
+#### features.enable_software_inventory
+
+Whether or not Fleet sends the query needed to gather the list of software installed on hosts, along with other metadata.
+
+- Optional setting (boolean).
+- Default value: `true`.
+- Config file format:
+  ```
+  features:
+  	enable_software_inventory: false
+  ```
+- Deprecated config file format:
+  ```
+  host_settings:
+  	enable_software_inventory: false
+  ```
+
+### Fleet Desktop Settings
+
+For more information about Fleet Desktop, see [Fleet Desktop's documentation](../../Using-Fleet/Fleet-desktop.md).
+
+#### fleet_desktop.transparency_url
+
+_Available in Fleet Premium_. Sets a custom transparency URL page to take users of Fleet Desktop to.
+
+- Optional setting (string).
+- Default value: Fleet's default transparency URL.
+- Config file format:
+  ```
+  fleet_desktop:
+    transparency_url: "https://example.org/transparency"
+  ```
+
+### Integrations
+
+For more information about integrations and Fleet automations in general, see the [Automations documentation](../../Using-Fleet/Automations.md). Only one automation can be enabled for a given automation type (e.g. for Failing Policies, only one of the webhook, the Jira integration or the Zendesk automation can be enabled).
+
+It is recommended to use the Fleet UI to configure integrations, as secret credentials (in the form of an API token) must be provided. See the [Automations documentation](../../Using-Fleet/Automations.md) for the UI configuration steps.
+
+#### Jira
```

Comment:
If the best practice is to configure integrations in the UI, I think we can remove the entire Jira and Zendesk section from the docs. This way the docs represent the best practice for using/configuring Fleet.

I think it makes sense to add the Jira and Zendesk section to the "Contributors" section to the docs and link to this section at the bottom of this page. 

With a warning like "More settings are including the the contributor docs. It's possible however not recommended to configure these settings in the `config` YAML."

What do you think?

Reply:
> I kinda like having all possible options documented

Ah, I see. I think we disagree here. 

I think we want the docs (and the product) to only include the best practice/recommended approach. This way, the user only needs to know/read to necessary material to get their job done (i.e. job of configuring integrations). 

So, if we recommend using the UI to control integrations then we should only document this.

@mna can you pass an environment variable to the `config` YAML? If yes, I think it makes sense to indicate that this is recommend in the docs.

If not, I'll stub out a section of the contributors docs. It makes sense that we don't recommend that the user stores API tokens in plain text.

Reply:
> can you pass an environment variable to the config YAML? If yes, I think it makes sense to indicate that this is recommend in the docs.

Tried it out, it doesn't support it, if we wanted to do that we'd need to hook it up in our yaml parsing or as a post-processing step AFAICT.

Reply:
I did move the Jira/Zendesk and SMTP settings to a new "Advanced Configuration" page in "Contributing". I left a TODO for you to add the introduction/context you had in mind, feel free to just write it in a comment too and I can add it to the PR proper.

Reply:
> I left a TODO for you to add the introduction/context you had in mind

Thanks! 

I'll add a commit directly to this PR and @mention you when I do this. 

Please let me know if you would prefer a comment.

Reply:
> I'll add a commit directly to this PR and @mention you when I do this.

Awesome, yeah that's fine!